sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/transforms/Combine.java

```diff
@@ -269,9 +282,22 @@ private Combine() {
    */
   public static <K, InputT, OutputT> GroupedValues<K, InputT, OutputT> groupedValues(
       PerKeyCombineFn<? super K, ? super InputT, ?, OutputT> fn) {
-    return new GroupedValues<>(fn);
+    return groupedValues(fn, getClassSafe(fn));
   }
 
+  private static <K, InputT, OutputT> GroupedValues<K, InputT, OutputT> groupedValues(
+          PerKeyCombineFn<? super K, ? super InputT, ?, OutputT> fn, Class<?> fnClass) {
+    return new GroupedValues<>(fn, fnClass);
+  }
+
+  private static Class<?> getClassSafe(Object fn) {
```

Comment:
I've converted this to use a new JavaClass type which is safe for serialization.